nixos/modules/services/backup/bupstash.nix

```diff
@@ -0,0 +1,666 @@
+                          '';
+                        };
+
+                        uri = mkOption {
+                          type = str;
+                          description = ''
+                            URI for the bupstash repository, can be a local path or in the form `ssh://$SERVER/$PATH`,
+                            will be used if the repository kind is `uri`.
+                          '';
+                        };
+
+                        ssh = {
+                          user = mkOption {
+                            type = str;
+                            description = "User used to connect to the remote server.";
+                          };
+
+                          server = mkOption {
+                            type = str;
+                            description = "Address ot the remote server.";
+                          };
+
+                          path = mkOption {
+                            type = nullOr str;
+                            default = null;
+                            description = ''
+                              Remote path for the repository.
+                              If no path is specified, then the ssh connexion to the remote server will not pass the `bupstash serve ...` command,
+                              instead relying that a `bupstash-serve` instance is directly available through ssh.
+                              This is the case if the remote repository has setup SSH access control with a custom `command`, as is the default with the NixOS module.
+                            '';
+                          };
+
+                          privateKey = mkOption {
+                            type = path;
+                            description = "Private SSH key used to connect to the server.";
+                          };
+                        };
+
+                        writePaths = mkOption {
+                          type = listOf path;
+                          default = [ ];
+                          description = "List of paths that the job can write to, useful for local repositories.";
+                        };
+
+                        command = mkOption {
+                          type = str;
+                          description = ''
+                            A command to run to connect to an instance of `bupstash-serve`.
+                            This allows more complex connections to the repository for less common use cases.
+                          '';
+                        };
+
+                        createRepository = mkOption {
+                          type = bool;
+                          default = false;
+                          description = ''
+                            Automatically creates the repository if not already done.
+                            Will cause the job to fail if activated and the repository already exists.
```

Comment:
This implies that the user would have to first do a run with the option enabled and then disable it for the subsequent runs. It would be preferable for the option to be idempotent, i.e. that once a repo is created this is ignored.

Reply:
The wording is not precise enough I think, what happens is : 
- If no repo exists and we set createRepo to true, then it will init a repo and remember it
- If the repo already exists before the first run (e.g, a common repo for multiple backups) then when trying to create it it will fail